pkg/planner/core/indexmerge_path.go

```diff
@@ -794,9 +816,35 @@ func (ds *DataSource) generateMVIndexMergePartialPaths4And(normalPathCnt int, in
 				for _, accessF := range accessFilters {
 					usedAccessCondsMap[string(accessF.HashCode())] = accessF
 				}
-				mvAndPartialPath = append(mvAndPartialPath, partialPaths...)
+				partialPaths4ThisMvIndex = append(partialPaths4ThisMvIndex, partialPaths...)
 			}
 		}
+		// sort allHashCodes to make the unified hashcode for slices of all accessFilters.
+		slices.Sort(allHashCodes)
+		allHashCodesKey := strings.Join(allHashCodes, "")
+		if rec, ok := mm[allHashCodesKey]; !ok {
+			// compute the count after access from those intersection partial paths, for this mv index usage.
+			countAfterAccess := float64(histColl.RealtimeCount) * cardinality.CalcTotalSelectivityForMVIdxPath(histColl, partialPaths4ThisMvIndex, true)
+			mm[allHashCodesKey] = &record{idx, partialPaths4ThisMvIndex, countAfterAccess}
+		} else {
+			// pick the minimum countAfterAccess's paths.
+			countAfterAccess := float64(histColl.RealtimeCount) * cardinality.CalcTotalSelectivityForMVIdxPath(histColl, partialPaths4ThisMvIndex, true)
+			if rec.countAfterAccess > countAfterAccess {
+				mm[allHashCodesKey] = &record{idx, partialPaths4ThisMvIndex, countAfterAccess}
+			}
+		}
+	}
+	// after all mv index is traversed, pick those remained paths which has already been de-duplicated for its accessFilters.
+	recordsCollection := make([]*record, 0, len(mm))
+	for _, one := range mm {
+		recordsCollection = append(recordsCollection, one)
+	}
```

Comment:
```suggestion
	recordsCollection := maps.Values(mm)
```
And import `"golang.org/x/exp/maps"`